### PR TITLE
feat:integrate bundled simulations

### LINKED ISSUE #229: Tenderly - Move from forks to bundled simulations
Currently we use forks in tenderly to fork the entire chain then run multiple simulations on that fork, this way we can save previous transactions states to accurately simulate an option in DAVI with multiple actions in order. (for example approving tokens that are spent in a spending call).

Tenderly just released an easier way of doing this with bundled simulations. We can do the same thing but without creating individual forks with complex logic we have now since its specifically designed for our use case.
Example in the docs here:
https://docs.tenderly.co/simulations-and-forks/simulation-api/bundled-simulations

It should also make it easier to show URLs hopefully but we still need to work out the exact URL structure for that. 
Doing #148 alongside this may be a good idea. (should not be done before this!)

**Acceptance criteria:**
We should test this against production since the expected result is the exact same outcome for both simulations if working correctly. 
Best test will be to do a vesting stream on gnosis chain where we try creating a stream with an approved number of tokens that we have in the guild and then another where we set the number of approved tokens to less than is being spent in the vesting creation. The first test should pass and the next should fail. (message Ross for questions regarding how exactly to do this test if unsure)

### PR DESCRIPTION
**Pull request by yathindrak**
Closes: https://github.com/DXgovernance/davi-monorepo/issues/229, https://github.com/DXgovernance/davi-monorepo/issues/148

- This integrates tenderly simulation bundles.
- Display a pointer to the relevant simulation results URL to dive deep into the simulation.

Screenshot 2023-03-03 at 23 52 23

### CODE REVIEW

**Comment by rossneilson** [comment]
Couple of findings, mostly it works amazing! and a lot faster than forks!

We used to highlight failed simulations with red borders, would be nice to get that back
Screenshot 2023-03-06 at 18 44 08

Then we don't show the failed message and link on the second action which is the one that failed (failed since we spent all the 1554 wxdai on the first action)
Screenshot 2023-03-06 at 18 45 10

Also I think we could slim the tenderly link section down to just one line saying: "Simulation [passed || failed] Tenderly logo<link>"

**Comment by yathindrak** [comment]
> Also I think we could slim the tenderly link section down to just one line saying: "Simulation [passed || failed] Tenderly logo"

Thanks, @rossneilson for pointing out the issues and improvements. By the way, the red border was appearing when I create an action for vesting with DXD. Will follow the same steps and find the reason behind. 

Screenshot 2023-03-06 at 19 12 11

Then slimming the tenderly link section down to just one line will be great to save the space for the simulation results block. Will check this as well.

**Comment by yathindrak** [comment]
@rossneilson The issue is fixed now and improvement is also done. Great if you could check now. :)
Screenshot 2023-03-06 at 20 59 28

**Comment by rossneilson** [comment]
> @rossneilson The issue is fixed now and improvement is also done. Great if you could check now. :) Screenshot 2023-03-06 at 20 59 28

Looks fantastic!
Could we maybe add in the error message from tenderly as well, pretty sure we had that before

Also looks like margin and padding might be a little off here
Screenshot 2023-03-07 at 13 05 06

**Comment by yathindrak** [comment]
Improvements are done and incresed the space between status text and tenderly link

Screenshot 2023-03-07 at 19 47 52

File: apps/davi/src/hooks/Guilds/useTenderlyApi/index.ts [hunk 1]
```diff
@@ -1,146 +1,118 @@
...
-import { Call, Option } from 'components/ActionsBuilder/types';
-import { useNetwork, useProvider } from 'wagmi';
+import { Option } from 'components/ActionsBuilder/types';
+import { useNetwork } from 'wagmi';
 
 // Variables are hardcoded because they didn't work
 // with GitHub. It might change later
 const TENDERLY_USER = 'dxdao';
 const TENDERLY_PROJECT = 'dxdao-proposal-simulation';
 const TENDERLY_ACCESS_KEY = 'fbzJG0PD3R8sX5i2vCCqh4r3YyrELtIO';
```
Comment (mprasanjith) [inline apps/davi/src/hooks/Guilds/useTenderlyApi/index.ts:8]: @rossneilson Should we move these to environment variables instead?

**Review by mprasanjith — APPROVED** [review]

**Review by rossneilson — APPROVED** [review]
Tested with multiple spending calls in a row and with multiple options, works great the same as current production

**Comment by Filipv95** [comment]
@yathindrak sry for the delay, jsut tested this PR - looks good to me, great job!

**Pull request merged**